src/modules/sub-items/components/pagination/pagination.component.js

```diff
@@ -0,0 +1,129 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import PaginationBtnComponent from './pagination.btn.component';
+
+export default class PaginationComponent extends PureComponent {
+    getPagesNumber() {
+        const { totalCount, itemsPerPage } = this.props;
+
+        return Math.ceil(totalCount / itemsPerPage);
+    }
+
+    /**
+     * Computes array with pagination pages.
+     * 
+     * Example 1: [ 1, "...", 5, 6, 7, 8, 9, 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 7)
+     * Example 2: [ 1, 2, 3, 4, 5, 6, 7, "...", 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 5) 
+     * Example 3: [ 1, "...", 8, 9, 10, 11, 12, "...", 20 ] (for: proximity = 2; pagesNumber = 20; currentPage = 10) 
+     *
+     * @returns {Array}
+     */
+    computePages() {
+        const { proximity, currentPage } = this.props;
+        const pagesNumber = this.getPagesNumber();
+
+        let startPage = Math.max(currentPage - proximity, 1);
+        let endPage = Math.min(currentPage + proximity, pagesNumber);
+
+        const rangeFromStartToEnd = [...Array(endPage - startPage + 1).keys()].map((i) => i + startPage);
+        const pages = rangeFromStartToEnd;
+
+        if (startPage > 3) {
+            pages.unshift(1, '...');
+        } else if (startPage === 3) {
+            pages.unshift(1, 2);
+        } else if (startPage === 2) {
+            pages.unshift(1);
+        }
+
+        if (endPage <= pagesNumber - 3) {
+            pages.push('...', pagesNumber);
+        } else if (endPage === pagesNumber - 2) {
+            pages.push(pagesNumber - 1, pagesNumber);
+        } else if (endPage === pagesNumber - 1) {
+            pages.push(pagesNumber);
+        }
+
+        return pages;
+    }
+
+    renderPrevious() {
+        const { currentPage, onPageChange } = this.props;
+        const previousPage = currentPage - 1;
+        const hasPreviousPage = currentPage !== 1;
+
+        return (
+            <PaginationBtnComponent
+                page={previousPage}
+                label="Back"
+                additionalClasses="prev"
+                disabled={!hasPreviousPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderNext() {
+        const { currentPage, onPageChange } = this.props;
+        const nextPage = currentPage + 1;
+        const hasNextPage = currentPage !== this.getPagesNumber();
```

Comment:
The same story like in `hasPreviousPage`. `isDisabled` or `isLastPage`, `currentPage === this.getPagesNumber()`